e2e/testcases/oci_sync_test.go

```diff
@@ -522,33 +521,49 @@ func testDigestUpdate(nt *nomostest.NT, image string) {
 }
 */
 
-func getImageDigest(imageName string, publicImage bool) (string, error) {
-	var auth authn.Authenticator
-	if publicImage {
-		auth = authn.Anonymous
-	} else {
-		a, err := google.NewEnvAuthenticator()
-		if err != nil {
-			return "", err
-		}
-		auth = a
-	}
-	image, err := oci.PullImage(imageName, remote.WithContext(context.TODO()), remote.WithAuth(auth))
+// getImageDigest uses gcloud to read the image digest of the specified image.
+// Using gcloud, instead of the OCI SDK, allows authenticating with local user
+// auth OR default app credentials. gcloud can authenticate with Google Artifact
+// Registry and Google Container Registry or pull with anonymous auth.
+// This allows reading the image digest without pulling the whole image.
+// Requires a sha256 image digest.
+func getImageDigest(nt *nomostest.NT, imageName string) (string, error) {
+	args := []string{
+		"gcloud", "container", "images", "describe",
+		imageName,
+		"--format", "value(image_summary.digest)",
+		"--verbosity", "error", // hide the warning about using "latest"
+	}
+	if *e2e.Debug {
+		nt.T.Log(strings.Join(args, " "))
+	}
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Env = os.Environ()
+	out, err := cmd.CombinedOutput()
 	if err != nil {
+		nt.T.Log(string(out))
 		return "", err
 	}
-
-	// Determine the digest of the image that was extracted
-	imageDigestHash, err := image.Digest()
-	if err != nil {
-		return "", err
+	hex, found := cutPrefix(strings.TrimSpace(string(out)), "sha256:")
+	if !found {
+		return "", fmt.Errorf("image %q has invalid digest %q", imageName, string(out))
 	}
-	return imageDigestHash.Hex, nil
+	return hex, nil
 }
 
-func imageDigestFunc(imageName string, publicImage bool) nomostest.Sha1Func {
-	return func(*nomostest.NT, types.NamespacedName) (string, error) {
-		return getImageDigest(imageName, publicImage)
+// imageDigestFunc wraps getImageDigest to return a Sha1Func that caches the
+// image digest, to avoid polling the image registry unnecessarily.
+func imageDigestFunc(imageName string) nomostest.Sha1Func {
+	var cached bool
+	var digest string
+	var err error
+	return func(nt *nomostest.NT, _ types.NamespacedName) (string, error) {
+		if cached {
```

Comment:
`cached` is always false. Do you want to make it a parameter of `imageDigestFunc`?

Reply:
why would it always be false?
It's set to true a few lines down.

Reply:
It should only be false the first time. Then it looks up the digest and sets it to true to skip digest lookup on subsequent calls.

Reply:
Nice optimization 👍

Reply:
Oh, right. The function is called in a retry loop and `cache` is set to true in the subsequent calls.